### DISCUSSION TITLE
SkiaSharp 2.88: Update uniforms of SKShader after creation?

### DISCUSSION THREAD

**Discussion by ziriax** [mono/SkiaSharp#2007] [Q&A]
Is it possible to update the uniform values of an SKShader after its creation?

**Comment by Costigan** [comment]
(Not an answer.)
I have the same question.  This should be possible, but API documentation for skiasharp seems to be limited to the API with little or no explanatory text, and tutorials are very hard to find.  Any pointers to an explanation of how to update the value of uniforms would be greatly appreciated.

**Reply by Costigan** [reply]
I have an example that appears to work well enough, but I don't have a mental model of the costs of doing this.

I'm creating a SKRuntimeEffect, a SKRuntimeEffectUniforms, a SKRuntimeEffectChildren and texture shader (from SKImage.FromEncodedData(<filename).ToShader) and holding on to these.  Then, every redisplay, I'm assigning values to the uniforms in the SKRuntimeEffectUniforms and then recreating and disposing of the shader I paint with using the SKRuntimeEffect's ToShader() method.  This appears to work quickly enough that I think the sksl code is getting compiled only once.

I still hope we can get an answer from someone who's working on the code.

**Comment by cyraid** [comment]
It would be nice to know, yeah. I'm worried when I have to create a shader all over again just when I update values. In GLSL uniforms are very quick, you get the uniform ID, and you can set the value directly using that id and poof, instantly updated.

Something like this would be nice if you have a `.Set()` from the `SKRuntimeEffectUniform`.

```csharp
SKRuntimeEffectUniform myFloat = 0.5f;
```
and later..
```csharp
myFloat.Set(1.0f);
```

Would be nice if that's all that was needed.
